### ISSUE TITLE
False `E0308` caused by wrong name resolution

### ISSUE THREAD

**Issue by z-jxy** [rust-lang/rust-analyzer#21668]
**rust-analyzer version**: 0.3.2795-standalone (00a9173e57 2026-02-15)

**rustc version**: rustc 1.92.0-nightly (9f32ccf35 2025-09-21)

**editor or extension**: Cursor (Extension version v0.3.2795)

**relevant settings**: -

**repository link (if public, optional)**: https://github.com/djeedai/bevy_hanabi (See additional context)

**code snippet to reproduce**:
```rust
// cargo add bevy_hanabi
use bevy_hanabi::prelude::*;

fn main() {
    let gradient = bevy_hanabi::Gradient::new();
    let modifier = ColorOverLifetimeModifier::new(gradient);
}
```

**Additional Context**:
This code compiles, but rust-analyzer is reporting an E0308 diagnostic `expected Gradient, found Gradient<{unknown}>`. 

When hovering `Gradient` in `ColorOverLifetimeModifier::new`'s signature in the dependency source, rust-analyzer shows `bevy_ui::gradients::Gradient` (an enum) instead of `bevy_hanabi::gradient::Gradient<T>`.

Based off source from [bevy_hanabi](https://github.com/djeedai/bevy_hanabi/blob/main/src/modifier/output.rs):
```rust
use bevy::prelude::*; // brings in `bevy_ui::gradients::Gradient`
use crate::Gradient; // brings in `bevy_hanabi::gradient::Gradient`

// ...

impl ColorOverLifetimeModifier {
    // rust-analyzer is reporting the type here as `bevy_ui::gradients::Gradient`
    fn new(gradient: Gradient<Vec4>) -> Self {
        // ...
    }
}
```

**Comment by wellscrosby** [comment]
Having this exact same issue.
